graphql_compiler/macros/macro_expansion.py

```diff
@@ -246,7 +250,7 @@ def _expand_macros_in_inner_ast(macro_registry, current_schema_type, ast, query_
         # No macro expansion happens at this level if there are no selections.
         return ast, query_args
 
-    macro_edges_at_this_type = macro_registry.macro_edges.get(current_schema_type.name, dict())
+    macro_edges_at_this_type = inherited_macro_edges.get(current_schema_type.name, dict())
```

Comment:
This looks like just a denormalization of the `macro_registry.macro_edges`. Any reason not to store it in the `macro_registry` itself? That way we know what was created from where, and we don't have to pass extra args.

Also, that would mean that we pay the denormalization cost once per macro edge definition at initialization, and not once per query on each run.

Reply:
I tried to be consistent with how we treat `schema_with_macros`. If we want to denormalize, I would denormalize both, in a separate PR.

Reply:
Sure, let's do that. Separate PR is fine.